samples/generated/express-web-with-oidc/server.js

```diff
@@ -0,0 +1,157 @@
+}
+
+const port = 8080;
+const app = express();
+
+app.use(express.static('./public'));
+app.use(express.urlencoded());
+app.post('/login', function(req, res) {
+  const issuer = req.body.issuer;
+  const username = req.body.username;
+  const password = req.body.password;
+  let status = '';
+  let sessionToken = '';
+  let error = '';
+  let authClient;
+
+  try {
+    authClient = new OktaAuthJS({
+      issuer
+    });
+  } catch(e) {
+    console.error('Caught exception in OktaAuthJS constructor: ', e);
+  }
+
+  authClient.signIn({
+    username,
+    password
+  })
+  .then(function(transaction) {
+    status = transaction.status;
+    sessionToken = transaction.sessionToken;
+    error = transaction.error;
+
+    const clientId = req.body.clientId;
+    const clientSecret = req.body.clientSecret;
+    const state = JSON.stringify({
+      username,
+      issuer,
+      clientId,
+      clientSecret
+    });
+    const baseUrl = issuer.indexOf('/oauth2') > 0 ? issuer : `${issuer}/oauth2`;
+    const authorizeUrl = `${baseUrl}/v1/authorize?` + querystring.stringify({
+      'client_id': clientId,
+      'response_type': 'code',
+      'scope': 'openid',
+      'prompt': 'none',
+      'redirect_uri': redirectUrl,
+      'state': state,
+      'nonce': uuid.v4(),
+      'sessionToken': sessionToken
+    });
+    res.redirect(authorizeUrl);
+
+  })
+  .catch(function(err) {
+    error = err;
+
+    // Return data to the client-side
+    const qs = querystring.stringify({
+      username,
+      issuer,
+      status,
+      error: error.toString(),
+    });
+    res.redirect('/?' + qs);
+  });
+});
+
+// Handle OIDC callback. The request query will contain a code and state
+app.get('/authorization-code/callback', function(req, res) {
+  // also known as "authorization_code"
+  const code = req.query.code;
+
+  // state can be any string. In this sample are using it to store our config
+  const state = JSON.parse(req.query.state);
+  const { issuer, clientId, clientSecret, username } = state;
+
+  const postData = querystring.stringify({
+    'grant_type': 'authorization_code',
+    'redirect_uri': redirectUrl,
+    'code': code
+  });
+  const baseUrl = issuer.indexOf('/oauth2') > 0 ? issuer : `${issuer}/oauth2`;
+  const encodedSecret = stringToBase64Url(`${clientId}:${clientSecret}`);
+  const post = https.request(`${baseUrl}/v1/token`, {
+    method: 'POST',
+    headers: {
+      'accept': 'application/json',
+      'authorization': `Basic ${encodedSecret}`,
+      'content-type': 'application/x-www-form-urlencoded',
+    }
+  }, (resp) => {
+    let data = '';
+
+    // A chunk of data has been recieved.
+    resp.on('data', (chunk) => {
+      data += chunk;
+    });
+
+    // The whole response has been received. Print out the result.
+    resp.on('end', () => {
+      const appUri = '/?'  + querystring.stringify(state);
+
+      res.send(`
+        <html>
+          <body>
+            <code id="accessToken">${data}</code>
```

Comment:
Based on the [spec for web app](https://oktawiki.atlassian.net/wiki/spaces/PM/pages/1047302310/Deployment+Models+for+Application+Development#Web-App), looks like accessToken should be stored on web server instead of being provided to the client-side.

Reply:
This is a very basic sample, and I'd like to show information in the UI both as a convenience and to help with our tests, which will not have access to the server-side information. I'm adding this to the HTML, how does this sound?

```
            <p>The OIDC flow succeeded. The access token is shown here as a convenience .<br/>
            In a real-world app, the access token should be stored server-side with the user's session.<br/>
            Leaking the access token to the client-side may represent a security risk if the token can be used to access protected resources.
            </p>

```